### PR TITLE
implemented InventoryFilters package

### PR DESCRIPTION
**Pull request by samvarankashyap**
InventoryFilters package has the abstract/concrete  classes needed for writing the filters. 
updated openstack, aws , gcloud inventory filters to implement object oriented InventoryFilter class.

### CODE REVIEW

File: provision/InventoryFilters/InventoryFilter.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        return inv['host_groups'].keys()
+
+    def get_layout_host_groups(self, inv):
+        return inv['host_groups'].keys()
+
+    def add_sections(self, config, section_list):
+        for section in section_list:
+            config.add_section(section)
+        return config
+    def set_children(self, config, inv):
```
Comment (herlo) [inline provision/InventoryFilters/InventoryFilter.py]: PEP8 - add spacing between methods.

File: provision/InventoryFilters/InventoryFilter.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        for host_group in inv['host_groups']:
+            if "children" in inv['host_groups'][host_group]:
+                config.add_section(host_group+":"+"children")
+                for child in inv['host_groups'][host_group]['children']:
+                    config.set(host_group+":"+"children",child)
+        return config
+    def set_vars(self, config, inv):
```
Comment (herlo) [inline provision/InventoryFilters/InventoryFilter.py]: PEP8

File: provision/InventoryFilters/InventoryFilter.py (hunk 1 above)
Comment (samvarankashyap) [inline provision/InventoryFilters/InventoryFilter.py]: i will run it through PEP8 tests and correct it.

**Comment by herlo** [comment]
Fix the PEP8 stuff and LGTM otherwise.

**Pull request merged**